tenacity/retry.py

```diff
@@ -111,6 +112,61 @@ def __call__(self, attempt):
             return not self.predicate(attempt.result())
 
 
+class retry_if_exception_message(retry_base):
+    """Retries if an exception message equals or matches."""
+
+    def __init__(self, message=None, match=None, require_exception=False):
+        if message and match:
+            raise TypeError(
+                "{}() takes either 'message' or 'match', not both".format(
+                    self.__class__.__name__))
+        self.predicate = self._predicate_factory(message, match)
+        self.require_exception = require_exception
+
+    def _predicate_factory(self, message, match):
+        if message:
+            def message_fnc(exception_message):
+                return message == exception_message
+            return message_fnc
+        elif match:
+            prog = re.compile(match)
+
+            def match_fnc(exception_message):
+                return prog.match(str(exception_message))
+            return match_fnc
+        raise TypeError(
+            "{}() missing 1 required argument 'message' or 'match'".format(
+                self.__class__.__name__))
+
+    def _no_failure(self):
+        """Retry logic for no failure."""
+        return False
+
+    def __call__(self, attempt):
+        if not attempt.failed:
+            if self.require_exception:
+                raise RuntimeError("No exception to get errormessage from")
+            return self._no_failure()
+        return self.predicate(attempt.exception().value)
+
+
+class retry_if_not_exception_message(retry_if_exception_message):
+    """Retries until an exception message equals or matches."""
+
+    def _predicate_factory(self, *args, **kwargs):
+        predicate = super(
+            retry_if_not_exception_message, self
+        )._predicate_factory(*args, **kwargs)
+
+        def inverted_predicate(*args_, **kwargs_):
+            return not predicate(*args_, **kwargs_)
+
+        return inverted_predicate
+
+    def _no_failure(self):
```

Comment:
@staticmethod

Not sure it's worth having a method, just an attribute would be enough.